test/ProtocolFeeController.t.sol

```diff
@@ -0,0 +1,378 @@
+    }
+
+    function testOwnerTransfer() public {
+        ProtocolFeeController controller = new ProtocolFeeController(address(clPoolManager));
+        // starts with address(this) as owner
+        assertEq(controller.owner(), address(this));
+
+        {
+            // must from owner
+            vm.prank(makeAddr("someone"));
+            vm.expectRevert(abi.encodeWithSelector(Ownable.OwnableUnauthorizedAccount.selector, makeAddr("someone")));
+            controller.transferOwnership(makeAddr("newOwner"));
+        }
+
+        controller.transferOwnership(makeAddr("newOwner"));
+
+        // still address(this) as owner before new owner accept
+        assertEq(controller.pendingOwner(), makeAddr("newOwner"));
+        assertEq(controller.owner(), address(this));
+
+        {
+            // must from pending owner
+            vm.prank(makeAddr("someone"));
+            vm.expectRevert(abi.encodeWithSelector(Ownable.OwnableUnauthorizedAccount.selector, makeAddr("someone")));
+            controller.acceptOwnership();
+        }
+
+        vm.prank(makeAddr("newOwner"));
+        controller.acceptOwnership();
+        assertEq(controller.owner(), makeAddr("newOwner"));
+    }
+
+    function testSetProcotolFeeSplitRatio(uint256 newProtocolFeeSplitRatio) public {
+        ProtocolFeeController controller = new ProtocolFeeController(address(clPoolManager));
+
+        {
+            // must from owner
+            vm.prank(makeAddr("someone"));
+            vm.expectRevert(abi.encodeWithSelector(Ownable.OwnableUnauthorizedAccount.selector, makeAddr("someone")));
+            controller.setProtocolFeeSplitRatio(newProtocolFeeSplitRatio);
+        }
+
+        if (newProtocolFeeSplitRatio > controller.ONE_HUNDRED_PERCENT_RATIO()) {
+            vm.expectRevert(ProtocolFeeController.InvliadProtocolFeeSplitRatio.selector);
+            controller.setProtocolFeeSplitRatio(newProtocolFeeSplitRatio);
+        } else {
+            controller.setProtocolFeeSplitRatio(newProtocolFeeSplitRatio);
+            assertEq(controller.protocolFeeSplitRatio(), newProtocolFeeSplitRatio);
+        }
+    }
+
+    function testCLPoolInitWithoutProtolFeeController(uint24 lpFee) public {
+        lpFee = uint24(bound(lpFee, 0, LPFeeLibrary.ONE_HUNDRED_PERCENT_FEE));
+        PoolKey memory key = PoolKey({
+            currency0: currency0,
+            currency1: currency1,
+            hooks: IHooks(address(0)),
+            poolManager: clPoolManager,
+            fee: lpFee,
+            parameters: bytes32(0).setTickSpacing(10)
+        });
+        clPoolManager.initialize(key, Constants.SQRT_RATIO_1_1);
+
+        (,, uint24 actualProtocolFee, uint24 actualLpFee) = clPoolManager.getSlot0(key.toId());
+
+        assertEq(actualLpFee, lpFee);
+        assertEq(actualProtocolFee, 0);
+    }
+
+    function testCLPoolInitWithProtolFeeControllerFuzz(uint24 lpFee, uint256 newProtocolFeeRatio) public {
```

Comment:
do u think its possible if we add in binPoolManager version test too? thx

Reply:
sure